tcs/csp_solver_two_tank_tes.h

```diff
@@ -325,39 +276,103 @@ class C_csp_two_tank_tes : public C_csp_tes
 
     virtual double get_degradation_rate();  // s^-1
 
-	virtual void discharge_avail_est(double T_cold_K, double step_s, double &q_dot_dc_est, double &m_dot_field_est, double &T_hot_field_est);
+	virtual void discharge_avail_est(double T_cold_K, double step_s, 
+		double &q_dot_dc_est /*MWt*/, double &m_dot_field_est /*kg/s*/, double &T_hot_field_est /*K*/);
 
-	virtual void charge_avail_est(double T_hot_K, double step_s, double &q_dot_ch_est, double &m_dot_field_est, double &T_cold_field_est);
+	virtual void charge_avail_est(double T_hot_K, double step_s, 
+		double &q_dot_ch_est /*MWt*/, double &m_dot_field_est /*kg/s*/, double &T_cold_field_est /*K*/);
```

Comment:
Is it still on the user to call these estimate functions before the discharge/charge functions?

Reply:
No, the remaining fluid in the tanks is calculated internally now, and they are not class members.